pandas/core/series.py

```diff
@@ -4559,6 +4560,98 @@ def replace(
+            """\
+            null_counts : bool, default True
+                Whether to show the non-null counts."""
+        ),
+        examples_sub=dedent(
+            """\
+            >>> int_values = [1, 2, 3, 4, 5]
+            >>> text_values = ['alpha', 'beta', 'gamma', 'delta', 'epsilon']
+            >>> s = pd.Series(text_values, index=int_values)
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            Series name: None
+            Non-Null Count  Dtype
+            --------------  -----
+            5 non-null      object
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Prints a summary excluding information about its values:
+
+            >>> s.info(verbose=False)
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Pipe output of Series.info to buffer instead of sys.stdout, get
+            buffer content and writes to a text file:
+
+            >>> import io
+            >>> buffer = io.StringIO()
+            >>> s.info(buf=buffer)
+            >>> s = buffer.getvalue()
+            >>> with open("df_info.txt", "w",
+            ...           encoding="utf-8") as f:  # doctest: +SKIP
+            ...     f.write(s)
+            260
+
+            The `memory_usage` parameter allows deep introspection mode, specially
+            useful for big Series and fine-tune memory optimization:
+
+            >>> random_strings_array = np.random.choice(['a', 'b', 'c'], 10 ** 6)
+            >>> s = pd.Series(np.random.choice(['a', 'b', 'c'], 10 ** 6))
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 7.6+ MB
+
+            >>> s.info(memory_usage='deep')
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 55.3 MB"""
+        ),
+        see_also_sub=dedent(
+            """\
+            Series.describe: Generate descriptive statistics of Series.
+            Series.memory_usage: Memory usage of Series."""
+        ),
+        version_added_sub="\n.. versionadded:: 1.2.0\n",
+    )
+    @doc(BaseInfo.render)
+    def info(
+        self,
+        verbose: Optional[bool] = None,
+        buf: Optional[IO[str]] = None,
+        max_cols: Optional[int] = None,
+        memory_usage: Optional[Union[bool, str]] = None,
+        null_counts: bool = True,
```

Comment:
per #36805 and #37999 , this should be `show_counts`

Reply:
Right, but one thing at a time. Will wait for the public API update first.